stingray/bexvar.py

```diff
@@ -0,0 +1,285 @@
+
+    def transform(cube):
+        params = cube.copy()
+        params[0] = cube[0] * (log_src_crs_grid[-1] - log_src_crs_grid[0]) + log_src_crs_grid[0]
+        params[1] = 10**(cube[1]*4 - 2)
+        return params
+
+    def loglike(params):
+        log_mean = params[0]
+        log_sigma = params[1]
+        # compute for each grid log-countrate its probability, according to log_mean, log_sigma
+        variance_pdf = scipy.stats.norm.pdf(log_src_crs_grid, log_mean, log_sigma)
+        # multiply that probability with the precomputed probabilities (pdfs)
+        likes = np.log((variance_pdf.reshape((1, -1)) * pdfs).mean(axis=1) + 1e-100)
+        like = likes.sum()
+        if not np.isfinite(like):
+            like = -1e300
+        return like
+
+    sampler = ReactiveNestedSampler(['logmean', 'logsigma'], loglike,
+    transform=transform, vectorized=False)
+    samples = sampler.run(viz_callback=False)['samples']
+    sampler.print_results()
+    log_mean, log_sigma = samples.transpose()
+
+    return log_sigma
+
+
+def bexvar(time, time_del, src_counts, bg_counts=None, bg_ratio=None, frac_exp=None):
+    """
+    Given a light curve data, Computes a Bayesian excess variance of count rate,
+    by estimating mean and variance of the log of the count rate.
+
+    Parameters
+    ----------
+    time : Iterable, `:class:numpy.array` or `:class:List` of floats, optional, default ``None``
+        A list or array of time stamps for a light curve.
+
+    time_del : iterable, `:class:numpy.array` or `:class:List` of floats
+        A list or array of time intervals for each bin of light curve.
+
+    src_counts : iterable, `:class:numpy.array` or `:class:List` of floats
+        A list or array of counts observed from source region in each bin.
+
+     bg_counts : iterable, `:class:numpy.array` or `:class:List` of floats, optional, default ``None``
+        A list or array of counts observed from background region in each bin. If ``None``
+        we assume it as a numpy array of zeros, of length equal to length of ``src_counts``.
+
+    bg_ratio : iterable, `:class:numpy.array` or `:class:List` of floats, optional, default ``None``
+        A list or array of source region area to background region area ratio in each bin.
+        If ``None`` we assume it as a numpy array of ones, of length equal the to length of
+        ``src_counts``.
+
+    frac_exp : iterable, `:class:numpy.array` or `:class:List` of floats, optional, default ``None``
+        A list or array of fractional exposers in each bin. If ``None`` we assume it as
+        a numpy array of ones, of length equal to length of ``src_counts``.
+
+    Returns
+    -------
+    posterior_log_sigma_src_cr : iterable, `:class:List` of floats
+        Contains an array of posterior samples of log(Sigma source count rate)
+        (i.e. log(Bayesian excess varience) of source count rates).
+    """
+
+    if bg_counts is None:

```

Comment:
Are these cases tested for? It is good to add a test case where these quantities are None, to test that, at least, the code runs without crashing

Reply:
Added tests for these cases.